nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,529 @@
+        wantedBy = [ "netbird-management.service" ];
+        restartTriggers = [ settingsFile managementFile ];
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package}/bin/netbird-signal run \
+              --port ${builtins.toString cfg.ports.signal} \
+              --log-file console \
+              --log-level ${cfg.logLevel}
+          '';
+          Restart = "always";
+          RuntimeDirectory = "netbird-mgmt";
+          StateDirectory = "netbird-mgmt";
+          WorkingDirectory = stateDir;
+        };
+        unitConfig = {
+          StartLimitInterval = 5;
+          StartLimitBurst = 10;
+        };
+        stopIfChanged = false;
+      };
+
+      systemd.services.netbird-management = {
+        description = "The management server for Netbird, a wireguard VPN";
+        documentation = [ "https://netbird.io/docs/" ];
+        after = [ "network.target" "netbird-setup.service" ];
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "netbird-signal.service" "netbird-setup.service" ];
+        restartTriggers = [ settingsFile managementFile ];
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package}/bin/netbird-mgmt management \
+              --config ${stateDir}/management.json \
+              --datadir ${stateDir}/data \
+              ${optionalString cfg.management.disableAnonymousMetrics "--disable-anonymous-metrics"} \
+              ${optionalString cfg.management.disableSingleAccountMode "--disable-single-account-mode"} \
+              --dns-domain ${cfg.management.dnsDomain} \
+              --single-account-mode-domain ${cfg.management.singleAccountModeDomain} \
+              --idp-sign-key-refresh-enabled \
+              --port ${builtins.toString cfg.ports.management} \
+              --log-file console \
+              --log-level ${cfg.logLevel}
+          '';
+          Restart = "always";
+          RuntimeDirectory = "netbird-mgmt";
+          StateDirectory = [ "netbird-mgmt" "netbird-mgmt/data" ];
+          WorkingDirectory = stateDir;
+        };
+        unitConfig = {
+          StartLimitInterval = 5;
+          StartLimitBurst = 10;
+        };
+        stopIfChanged = false;
+      };
+    })
+
+    (mkIf cfg.enableCoturn {
+      services.coturn = {
+        enable = true;
+
+        realm = settings.NETBIRD_DOMAIN;
+        lt-cred-mech = true;
+        no-cli = true;
+
+        extraConfig = ''
+          fingerprint
+
+          user=${settings.TURN_USER}:${builtins.toString settings.TURN_PASSWORD}
+          no-software-attribute
+        '';
+      };
+
+      networking.firewall = {
+        allowedUDPPorts = with settings; [ TURN_PORT (TURN_PORT + 1) 5349 5350 ];
+        allowedTCPPorts = with settings; [ TURN_PORT (TURN_PORT + 1) ];
+        allowedUDPPortRanges = [{ from = settings.TURN_MIN_PORT; to = settings.TURN_MAX_PORT; }];
+      };
+    })
+
+    (mkIf (cfg.enableNginx && cfg.enableCoturn) {
```

Comment:
There should be an additional option like `enableACME` to handle this. If used with a different reverse proxy this will not work

Reply:
I mean, the option is named `configureNginx`, as I don't use any other reverse proxy, I don't feel confident maintaining them. The best course of action is to set `configureNginx` to `false` and configure your reverse proxy yourself.

Reply:
Yes and no. There should be a nginx config to host the dashboard.
Independent of this there could be a config to also reverse proxy the other two services incl acme.
This way, I could have nginx as plain webserver but still use a different reverse proxy